### ISSUE TITLE
Idea: A mobile and analytics platform available to social justice groups and the general public which enables people to capture evidence of social injustice using a mobile app

### ISSUE THREAD

**Issue by cfc-emb-race** [Call-for-Code/Embrace-Judicial-Reform#4]
**Name:**
Enoch Antwi & James Stewart

**Theme:**
Policy and Judicial Reform

**Brief description of your idea:**
A mobile and analytics platform available to social justice groups and the general public which enables people to capture evidence of social injustice using a mobile app. The video / other evidence is timestamped & geo-tagged in order to collate evidence into groups pertaining to a single incident. Speech to text and other video analytics can also be used in order to capture transcripts and highlight any racially motivated language and potentially compare to a legal corpus / other cases in order to understand the implications and whether the incident should be reported. Features that alert others with the app to the location of an incident and key words to trigger recording (even just sound) could also be added.

**What makes your idea unique?:**
To our knowledge this has not been implemented previously. Provides the opportunity to capture long term information around patterns of behaviour, this information could be analysed in order to help with reform (policing, judicial, policies etc)

**What would be the impact of your idea if implemented?:**
Could prevent incidents from escalating, real time capture of possible injustices and evidence forming clear visibility of racially motived incidents.

**Skills to contribute (e.g. development, architecture, research, design or anything else):**
Real life experience, research, design

**Comment by bbbozzz** [comment]
A few suggestions:

1. People are going to use whatever app they are comfortable with.
2. Only a subset of those people will be willing to share on social media.
3. It should be possible to create a simple sharing mechanism that a) collects artifacts such as audio and video, and b) establishes the provenance of that audio and video. This can also be done in a privacy preserving way while also addressing the potential for abuse. 
4. A public data store for this kind of data could be useful on its own, but it also shares quite a few architectural similarities with public data stores from other sources such as police body cams.  
5. Thus it may make sense to work the larger issue of aggregating source information and developing high-quality metadata.
6. In keeping with recent scholarship, this may extend beyond police abuse to whether the right non-police resources are targeted to address a problem. It may be possible to quickly tie in video of an ongoing event so that 911 screeners could send additional help beyond police, such as psychologists, counselors, or an ambulance as an alternative.  

Hope that helps!

**Comment by Jamstew2** [comment]
@bbbozzz yes thanks for the input we will be doing some further thinking around this initial idea in the coming days so your feedback is gratefully received.

**Comment by tnadams** [comment]
I actually just submitted a disclosure for something very similar to this idea. I did talk to IBM IP and they said I am free to help contribute to this while that is still pending, but wanted to disclose that info up front.

I think this is a great idea, and agree with @bbbozzz #5 above that the issue of aggregation and authenticated meta-data is of utmost importance here. Using a 3rd party authentication system to verify GPS and time stamped data would help validate witness evidence for trial and investigative use. Meta-data can be changed after-the-fact, so I think having access to the source data at the time of the event is important for this process to work.

**Comment by davidzunigaibm** [comment]
Hi, I think this is a great idea, is there a need for a Data Analyst/Scientist in your team? 
I would like to contribute in any way I can.

You can reach me at: david.zuniga@ibm.com

Thank you!

**Comment by amfred** [comment]
I've had such an app on my phone for some time.  It's called Mobile Justice and it's developed in collaboration with the ACLU: 

https://apps.apple.com/us/app/mobile-justice-california/id979642692#see-all/developer-other-apps

It's limited to certain states for some reason, though.

**Comment by Jamstew2** [comment]
@amfred thanks was not aware of this app (but based in the UK so that may explain why). In addition to the application, we were hoping to build the video analytics capability, including speech to text and also capturing timestamps and geotag from the metadata in order to better collate evidence. Do you think these would be differentiators which may even support existing apps like Mobile Justice?

**Comment by amfred** [comment]
It looks like the app is already using the phone's location data as a geo-location tag, but the video analytics capabilities could be new.

**Comment by Jamstew2** [comment]
@amfred ok thanks, that is really helpful. Sounds like our strength could be at the "backend" in terms of 

- Identifying entities within a video
- Speech to text (so we can do transcripts and further analysis based on an NLU model)
- Potentially build models to help offer long term advice and guidance to both citizens and law enforcement on how to improve practices / training etc
- If possible perhaps there are still app features we could enhance / build (e.g. improve the ability to "stitch" videos together into a single "case file" based on metadata or the "trigger word" idea we had to start recording audio hands free if not already there).

**Comment by tnadams** [comment]
@amfred Thanks for the reference, looks like a lot of similarity. Do you know how the captured video is stored? Is it uploaded to a central repo or just stored locally on the user's phone? I think an important part of the process would need to be 3rd party verification of metadata, as metadata can be changed after-the-fact making it harder to use as evidence later. Here is a link with some of the difficulties in presenting video evidence in court: https://digitalcommons.tourolaw.edu/cgi/viewcontent.cgi?referer=https://www.google.com/&httpsredir=1&article=1228&context=lawreview. Any final solution would need to have some processes in place to hopefully meet these requirements.

**Comment by davidzunigaibm** [comment]
@Jamstew2 I have an idea for the point you make to create group reports for a "case file" involving the creation of event identifiers from date, time and location of the report. I could help creating a solution base in Python for this if is possible

**Comment by amfred** [comment]
I'm not sure @tnadams.  The ACLU is an organization dedicated to civil rights and specifically legal action, so they would presumably do a better job of this than most.  However, we'd have to reach out to them to get technical details.

One article says that the video is "emailed" to the state ACLU, but it may be that it's actually transferred in some other way.  Either way, it's instantly sent from your phone to the ACLU when you stop recording, and then you can add additional metadata to help with follow-up.

**Comment by amfred** [comment]
Resources: 
https://www.aclu.org/issues/criminal-law-reform/reforming-police/aclu-apps-record-police-conduct
https://www.cnn.com/2020/05/31/us/aclu-mobile-justice-police-misconduct-app/index.html

**Comment by Jamstew2** [comment]
@tnadams I think we would need to ensure the meta data captured was immutable and could not be changed once submitted. I'm sure there are a few ways to achieve this, but one I'm familiar with would be to use a Blockchain solution, there are techniques to store the actual images / video off-chain but use a salted hash which links to a record on the blockchain in order to ensure the record only links to the original image (otherwise we know it was tampered with in some way). That is just one method I've seen before (as I say there could be others). @DaveZuniga sounds good, it sounds like they are just using email today to collect the videos so perhaps an automated way to stitch / collate videos together would be good? To tnadams' point data sent via email could be tampered with.

**Comment by BchanceIBM** [comment]
@Jamstew2 this is a great idea and could be applied across a few of our problem statements - for example in Problem Statement #3 we could train an algo based on past courtroom sentencing recordings (stt and video) to identify judicial bias and then apply and evolve/use it in current courtroom cases to identify biased sentencing.

**Comment by helloannietran** [comment]
Hi! My name is Annie. I'm a data scientist from Red Hat. I think this is a cool idea and would love to collaborate if there's room on the team. I've done one project on image recognition using R-CNN and currently taking fastai courses, so I'm still relatively new to deep learning but would love an opportunity to apply what I learn to a project, especially one with impact. If there's space, would this be for problem statement 1?

**Comment by Jamstew2** [comment]
@helloannietran Thanks so much for getting in touch. We would love you to join our team. We have a data scientist onboard @DaveZuniga but I'm hoping you could work together on this? @helloannietran we have our team kickoff today so if you contact me directly (jamstew2@uk.ibm.com) I will add you to the Webex.

**Comment by davidzunigaibm** [comment]
@Jamstew2  Yes!, I don't have a lot of experience in the field, so the help would be really appreciated. I am happy to either to work with @helloannietran and/or take any other role if necessary.

**Comment by helloannietran** [comment]
Yes! I would love to collaborate with you, @DaveZuniga  :)

**Comment by MarkWoolnough** [comment]
Hiya - I work over in the Systems side of the house.  :-)  I can bring access to some sexy Power9 hardware, Auto AI and python/jupyter/pandas hackery plus a lot of experience working with data and analytics of various flavours.  Happy to roll my sleeves up here and get involved if its not too late.  Cheers! Mark
